myokit/formats/sbml/_importer.py

```diff
@@ -159,42 +188,101 @@ def model(self, path):
         # Return finished model
         return model
 
-    def _parse_initial_assignments(self, model, comp, refs, node):
+    def _get_namespaces(self, sbml_version: str, log):
+        """
+        Defines namespaces for supported SBML versions. Supported versions are:
+        level 2, version 3, 4, 5.
+        """
+        ns = dict()
+        if sbml_version == "{http://www.sbml.org/sbml/level2/version3}":
+            # SBML
+            ns['sbml'] = "{http://www.sbml.org/sbml/level2/version3}"
+            # MathML
+            ns['mathml'] = "{http://www.w3.org/1998/Math/MathML}"
+        elif sbml_version == "{http://www.sbml.org/sbml/level2/version4}":
+            # SBML
+            ns['sbml'] = "{http://www.sbml.org/sbml/level2/version4}"
+            # MathML
+            ns['mathml'] = "{http://www.w3.org/1998/Math/MathML}"
+        elif sbml_version == "{http://www.sbml.org/sbml/level2/version5}":
+            # SBML
+            ns['sbml'] = "{http://www.sbml.org/sbml/level2/version5}"
+            # MathML
+            ns['mathml'] = "{http://www.w3.org/1998/Math/MathML}"
+        elif sbml_version == "{http://www.sbml.org/sbml/level3/version1}":
+            # SBML
+            ns['sbml'] = "{http://www.sbml.org/sbml/level3/version1}"
+            # MathML
+            ns['mathml'] = "{http://www.w3.org/1998/Math/MathML}"
+        elif sbml_version == "{http://www.sbml.org/sbml/level3/version2}":
+            # SBML
+            ns['sbml'] = "{http://www.sbml.org/sbml/level3/version2}"
+            # MathML
+            ns['mathml'] = "{http://www.w3.org/1998/Math/MathML}"
+        else:
+            # SBML
+            ns['sbml'] = sbml_version
+            # MathML
+            ns['mathml'] = "{http://www.w3.org/1998/Math/MathML}"
+
+            # log import warning
+            log.warn('The SBML version %s has not been tested. The model'
+                     % sbml_version + ' may not be imported correctly.'
+                     )
+
+        return ns
+
+    def _get_sbml_version(self, root):
+        """
+        Returns the SBML version of the file.
+        """
+        m = re.match(r'\{.*\}', root.tag)
```

Comment:
What's happening here? Can we not get the version from either the namespace of the root element, or from the attributes of the root element?

Reply:
There doesn't seem to be a native function in etree to access the namespace of any variable. So what this does is it gets the tag of the root, i.e. the name/ id, and straps off it's namespace, which is the SBML version.

I haven't found a better way to do this.

Reply:
Oh ok! I see what this is doing now.
There's a method `mxml.split` that splits off the namespace in much the same way. Might be worth using that instead. (Not "better", just means we're using a single method consistently and can fix bugs more easily etc)

Reply:
Done